### ISSUE TITLE
Contant Time Trajectory with C++ API

### ISSUE THREAD

**Issue by JadTawil-theonly** [hungpham2511/toppra#134]
Hello,

Is an analog for the set_desired_duration method available in the c++ implementation?

**Comment by JadTawil-theonly** [comment]
Also, it seems like the c++ API requires paths which are composed of segments which are parametrized as polynomials. The python API seems to accept arbitrary waypoints, as opposed to waypoints generated by piecewise polynomials. Is this true? And why so?

**Comment by hungpham2511** [comment]
Yes the C++ API as of now only support polynomial path, but you can implement your own path types by inheriting from the abstract interface.

> Is an analog for the set_desired_duration method available in the c++ implementation?

This functionality is not available in the C++ impl.

**Comment by lbylbylbyl** [comment]
> Also, it seems like the c++ API requires paths which are composed of segments which are parametrized as polynomials. The python API seems to accept arbitrary waypoints, as opposed to waypoints generated by piecewise polynomials. Is this true? And why so?
hi, do you find or make the c++ use of toppra?can you share me with it?i really need it,thanks

**Comment by hungpham2511** [comment]
The python api also constructs a piecewise polynomial internally.
